### DISCUSSION TITLE
Please change distribution process

### DISCUSSION THREAD

**Discussion by julian-weinert** [ExistentialAudio/BlackHole#484] [General]
I get how you did it in the beginning, even though I strongly disagree.
But by now it's just nothing but a road block.

We are supposed to enter our personal details and email *every time* there is an update.
And since a few months I can't get it anymore because I don't receive a download link by mail.

I think it's time to grow up as project and implement a proper, public, distribution strategy.
Come across less fishy and more official.

What do you think?

**Comment by devinroth** [comment]
Hey Julian. What do you suggest?

Major issues are that I have very little time (3 kids and one is 6 months) and it’s not like open source pays.

Devin

**Reply by julian-weinert** [reply]
I would suggest the GitHub releases. You would simply provide a zip or DMG file next to the source code.

The other option is ridiculously obvious. Put the download link on the website, just as everybody else does it.

You have my name and e-mail 10 times now and I wonder what you want that data for.

Because the process of sending out mails is far more complex than simply uploading the file to the release...

Now I'm not receiving the mail at all after 20 minutes and all other mail hosts I could try got blocked by you in the meantime

**Comment by devinroth** [comment]
Why not just use homebrew?

Devin

**Reply by Feanux3** [reply]
Speaking of homebrew, would it be feasible to allow installing the same driver but named differently? Similar to how you have to do it when you build the driver yourself. I know I can install 2ch and 16ch side by side, but having two 2ch drivers installed via homebrew would resolve a lot of signing headaches I'm running in to when building them.

**Comment by vlad-cotfas** [comment]
I am also in this situation, maybe if the driver also runs in the background, why not do periodically checkups (this must be controllable, as some of ppl have low internet and hotspot), and when an update came up to do a HUD or a simple popup, I think there is also a library which does this. As it is open source, we can get the code and compile it, when we have time, do PRs and the owner can approve the PRs on the commit, we also have to give back to the community, not only to take.

I can help you with the code for HUD for example, HUD is the item that shows on the screen when you volume up/down, and I have the code, I use it internally for other purposes for myself, this is against.. it is using private APIs so... :S, a notification will be easier, BUT, you can also check the notifications all from github and you can receive as well emails from github when the release is made. So yeah, release tab in github will be good. (I wanted yesterday to update and I did it with brew install..)

Beside that GG!

**Reply by orchetect** [reply]
If BlackHole makes it onto the App Store one day, we can't rely on private APIs however.

**Reply by julian-weinert** [reply]
If it does this whole thing is out the window anyway.
As long as you track the use of private APIs that would be fine. If the updater code includes those it's simple to get rid of, because the updater wouldn't be needed at all anymore.

**Reply by julian-weinert** [reply]
But I don't really think that would require the use of private APIs anyway.

**Comment by devinroth** [comment]
I like that idea.

Sparkle might do the trick.

Devin

**Reply by vlad-cotfas** [reply]
Yes, Sparkle was the library, easier to mantain, for me as a user, I do not want to spend to much time on unrelated tasks (updating apps)

**Comment by julian-weinert** [comment]
Oooooh! I really like that too!
I mean I guess it's more work on your part, but it's an even better solution than I had wished.
Sorry if my response the other day sounded a bit rude, I was just super tired and stressed and didn't phrase kindly

**Comment by julian-weinert** [comment]
Just to get into the topic again.
Sparkle would be the way to go. But that's future things.

What I'm inquiring about is simply putting the disk image you mail a link to on the GitHub releases page.

I couldn't install BlackHole updates since I posted this thing, since your e-mail system seems to decline my e-mail address.
It's just no proper way to distribute apps beyond a beta phase

**Reply by devinroth** [reply]
It’s the other way. Your email system is refusing my emails. Just use something like email-fake.com.
